src/ASM/ASMs2DLag.C

```diff
@@ -826,15 +826,25 @@ bool ASMs2DLag::write(std::ostream& os, int) const
 int ASMs2DLag::findElement(double u, double v,
                            double* xi, double* eta) const
 {
-  int elmx = std::min(nx-2.0, floor(u*(nx-1)));
-  int elmy = std::min(ny-2.0, floor(v*(ny-1)));
+  int ku = surf->basis(0).knotInterval(u);
+  int kv = surf->basis(1).knotInterval(v);
+  int elmx = ku - (p1 - 1);
+  int elmy = kv - (p2 - 1);
+
+  if (xi) {
+      const double knot_1 = *(surf->basis(0).begin() + ku);
+      const double knot_2 = *(surf->basis(0).begin() + ku+1);
+      *xi   = -1.0 + (u - knot_1) / (knot_2-knot_1) * 2.0;
+  }
+  if (eta) {
+    const double knot_1 = *(surf->basis(1).begin() + kv);
+    const double knot_2 = *(surf->basis(1).begin() + kv+1);
+    *eta  = -1.0 + (v  - knot_1) / (knot_2-knot_1) * 2.0;
```

Comment:
only one space in `(v - knot_1)`. Maybe also space around the `-` in the denominator, fr consistency (also in L837)

Reply:
Another (cosmetic) thing: When having expressions like `a + b / c * 2.0` I always have to think twice which of the operators `*` and `/` will be evaluated first (I guess they are always from the left, i.e., the `/` first and then `*`), since it will affect the result if the `*` is evaluated first. Therefore, suggest rewriting to the equivalent `a + 2.0 * b / c` and there will be no such worries.

Reply:
fixed, fixed, fixed.